console/v1/types_console_notification.go

```diff
@@ -0,0 +1,54 @@
+package v1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Extension for configuring openshift web console notifications.
+type ConsoleNotification struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object's metadata.
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              ConsoleNotificationSpec `json:"spec"`
+}
+
+type ConsoleNotificationSpec struct {
+	// text is the visible text of the notification
+	Text string `json:"text"`
+	// location is the location of the notification in the console
+	// +optional
+	Location ConsoleNotificationLocation `json:"location,omitempty"`
+	// link is an object that holds notification link details
+	// +optional
+	Link ConsoleLink `json:"link,omitempty"`
```

Comment:
I don't think this can be omitempty unless it's a pointer

Reply:
Right, it won't error, but it would default it to `link: {text:"", href:"" }` instead of dropping it. I'll update to a pointer.